File: ch3/reverse_string.rb
def rev(theString)
  if theString.length == 0 || theString.length == 1
    # BASE CASE
    return theString
  else
    head = theString[0]
    tail = theString[1..]
    return rev(tail) + head
  end
end

puts(rev("abcdef")) # => fedcba
puts(rev("Hello, world!")) # => !dlrow ,olleH
puts(rev("")) # =>
puts(rev("X")) # => X

File: ch3/reverseString.py
def rev(theString):
  if len(theString) == 0 or len(theString) == 1:
    # BASE CASE
    return theString
  else:
    # RECURSIVE CASE
    head = theString[0]
    tail = theString[1:]
    return rev(tail) + head

print(rev("abcdef")) # => fedcba
print(rev("Hello, world!")) # => !dlrow ,olleH
print(rev("")) # =>
print(rev("X")) # => X